diff -Naur sharutils-4.15/src/shar.c sharutils-4.15.tpg/src/shar.c
--- sharutils-4.15/src/shar.c	2015-01-07 02:11:42.000000000 +0000
+++ sharutils-4.15.tpg/src/shar.c	2015-02-21 00:19:58.689909120 +0000
@@ -931,7 +931,7 @@
         free (c_dir);
       }
     else
-      error (0, errno, _("Cannot get current directory name"));
+      error (0, errno, "%s", _("Cannot get current directory name"));
   }
 }
 
@@ -2154,7 +2154,7 @@
        */
       FILE * fp = freopen ("/dev/null", fwriteonly_mode, stderr);
       if (fp != stderr)
-        error (SHAR_EXIT_FAILED, errno,
+        error (SHAR_EXIT_FAILED, errno, "%s",
                _("reopening stderr to /dev/null"));
     }
 
@@ -2225,7 +2225,7 @@
   if (HAVE_OPT(QUERY_USER))
     {
       if (HAVE_OPT(NET_HEADERS))
-	error (0, 0, _("PLEASE avoid -X shars on Usenet or public networks"));
+	error (0, 0, "%s", _("PLEASE avoid -X shars on Usenet or public networks"));
 
       fputs ("shar_wish=\n", output);
     }
